ray-operator/controllers/raycluster_controller.go

```diff
@@ -501,3 +513,168 @@ func (r *RayClusterReconciler) updateStatus(instance *rayiov1alpha1.RayCluster)
 
 	return nil
 }
+
+func (r *RayClusterReconciler) reconcileServiceAccounts(instance *rayiov1alpha1.RayCluster) error {
+	if instance.Spec.EnableInTreeAutoscaling == nil || !*instance.Spec.EnableInTreeAutoscaling {
+		return nil
+	}
+
+	serviceAccounts := corev1.ServiceAccountList{}
+	filterLabels := client.MatchingLabels{common.RayClusterLabelKey: instance.Name}
+	if err := r.List(context.TODO(), &serviceAccounts, client.InNamespace(instance.Namespace), filterLabels); err != nil {
```

Comment:
A little question, we might use `r.Get` to get resource object if the resource name equals to raycluster name?